src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -14,14 +14,17 @@
 
 from __future__ import annotations
 
+import inspect
 from functools import wraps
 from typing import Any, Dict, Optional, Union
 
+import typer
 from click import ClickException
 from snowflake.cli.api.cli_global_context import (
     get_cli_context,
     get_cli_context_manager,
 )
+from snowflake.cli.api.commands.decorators import _options_decorator_factory
```

Comment:
making this function public would require CLI team review, and I'm not sure we want to do that anyways